app/models/effective/page.rb

```diff
@@ -1,5 +1,39 @@
 module Effective
   class Page < ActiveRecord::Base
+    if defined?(PgSearch)
+      include PgSearch::Model
+
+      pg_search_scope :search,
+                      against: [
+                        :title,
+                        :menu_title,
+                        :meta_description,
+                        :slug,
+                      ],
+                      associated_against: {
+                        rich_texts: [:body],
+                      },
+                      using: { tsearch: { highlight: true }, trigram: { word_similarity: true } }
+
+      multisearchable against: [
+                        :title,
+                        :menu_title,
+                        :meta_description,
+                        :slug,
+                      ],
+                      associated_against: {
+                        rich_texts: [:body],
+                      },
+                      using: {
+                        trigram: {},
+                        tsearch: {
+                          highlight: true,
+                        }
+                      },
+                      ranked_by: ":trigram", # Could rank by any column/expression, e.g.: (books.num_pages * :trigram) + (:tsearch / 2.0)
+                      if: -> (page) { page.published? }
```

Comment:
Does this get considered whenever a search is performed?  Or is this when the record is updated?

The published? is a Time.zone.now under the hood. If it's considered at search time, this is really great.  If it's considered at save time, this might not work as expected.

Reply:
Only when pg search documents are created/updated

That is a great question, I noticed that `Page#published?` just considers the draft column, then I used the conditional but for `Effective::Post#published?` it also takes into account the `Time.zone.now` so I didn't use it there - how you think we should do?

Reply:
Yea, maybe in search results we need to consider the published? status then